src/regularized/regularized.jl

```diff
@@ -0,0 +1,46 @@
+"""
+optimizer: θ ⟼ argmax θᵀy - Ω(y)
+"""
+struct Regularized{O,R}
```

Comment:
Do we also need the linear maximizer as a field for when the layer is called outside of training?
It would make sense to me to modify the behavior of `Perturbed` as well so that the standard forward pass just calls the naked linear maximizer